### ISSUE TITLE
Accessible solution needed for making the challenges completed on the map

### ISSUE THREAD

**Issue by benmcmahon100** [FreeCodeCamp/FreeCodeCamp#2851]
@zersiax pointed out to me that we have marked challenges as completed challenges on the map as complete by graying them out which means people of limited or no ability to see can't decipher whether or not they are complete.

**Comment by zersiax** [comment]
Proposed fix:
Add a screenreader-only span element to the loop showing all the challenges that only shows up when a challenge is complete, stating 'completed'. SR-only classes can be gleaned from the Bootstrap CSS source if you need an example, but generally take the form of a 1px by 1px element or an element that is visually rendered off screen so it doesn't screw up the rest of the design. I'd do it myself and PR, but I'm not familiar enough with the FCC source to know where this fixwould need to be implemented.

**Comment by benmcmahon100** [comment]
@BerkeleyTrue don't suppose you know how to do this?

**Comment by BerkeleyTrue** [comment]
There is a bootstrap class `sr-only` that can be added to a second filtered list.

**Comment by zersiax** [comment]
which looks like this:
.sr-only {
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    margin: -1px;
    overflow: hidden;
    clip: rect(0,0,0,0);
    border: 0;
}

I quickly glanced through the .less files in the FCC staging branch and saw no such similar class, therefore I went hunting and tracked down the code needed real quick. Any element that has this CSS class is invisible on screen and only detectable by screenreaders

**Comment by BerkeleyTrue** [comment]
https://github.com/FreeCodeCamp/FreeCodeCamp/blob/306c441ae75c2a0cc7decdfa0677c8ffdf7dc83f/client/less/lib/bootstrap/scaffolding.less#L125

**Comment by BerkeleyTrue** [comment]
Yes it would be in the html twice

**Comment by zersiax** [comment]
You wouldn't want to do that then, since a screenreader would have the page rendered twice. Once regularly, and once only the completed challenges. I think adding a bit of text to the completed items instead would be both more user friendly and more performant in the long run, no?

**Comment by BerkeleyTrue** [comment]
performance is not a big issue here. 

Not sure what is the best option. Not sure if screen readers render both list. If that would make sr-only pointless. 

Would the sr-only list being first alleviate this?

**Comment by benmcmahon100** [comment]
What about an invisible span after each challenge with complete or incomplete in it

**Comment by zersiax** [comment]
It might. But hear me out. Pseudocode:
Foreach(challenge in challenges)
if(completed) {
Render(inGrey); // function to show one particular challenge
renderElement(<span class=sr-only">completed</span>); // just create a vanilla  HTML element
} else{
Render(regular) // code for rendering challenge non-greyed out

This code makes the asumption that to generate the challenge map the list is iterated through at some point. If it is not, please tell me how the mechanism works so I can think of a more suitable solution. I guess rendering the completed challenges before the rest works, but is still a little cumbersome since you'd have to jump from the 'completed'  list to the 'regular'  list everytime you want to switch from checking what you've finished to actually going to the next challenge

**Comment by zersiax** [comment]
hahaha thats what I've been saying the whole time ;)

**Comment by benmcmahon100** [comment]
If that span thing is okay then it should be just
"<li class = \""+ challenge.status + "\">" + challenge.name + "<span class = \"sr-only\">" + challenge.status + "</span></li>"
Except I have no idea where this is located :p

**Comment by BerkeleyTrue** [comment]
in the jade

**Comment by benmcmahon100** [comment]
Cool I'll fix and pr now! We still working off of staging?

**Comment by zersiax** [comment]
once I'm a bit more well-versed I'll try to give back to the FCC codebase but for now, I think I'll break more than I fix

**Comment by BerkeleyTrue** [comment]
yes

**Comment by benmcmahon100** [comment]
Okay so I've added it to staging and it doesn't render as viewable text but I've just realized that i have no idea on how to test whether or not the screen reader works

**Comment by zersiax** [comment]
easiest way to check is to check your HTML source. if it shows up there, the screenread

**Comment by zersiax** [comment]
Or ..well I could test for you but I can't get at your testing area ;)

**Comment by benmcmahon100** [comment]
You can clone it if you had the time. Other wise we'll find out when it's deployed
here's the html if it's any use 
![image](https://cloud.githubusercontent.com/assets/10749142/9564207/6fe9daf0-4e96-11e5-864a-5f06ea1b68f3.png)

**Comment by zersiax** [comment]
You do know you just sent a screenshot to a blind person, ... right? :P Anyway I'll try to clone and setup, is the readme.md up to date on setting up now?

**Comment by benmcmahon100** [comment]
Well that is officially the stupidest thing I have ever done in my life. The guide should be accurate! I'm gonna go and find a wall to bash my head off of

**Comment by benmcmahon100** [comment]
Sometimes I amaze even myself

**Comment by zersiax** [comment]
chuckles :) its all good, people tend to forget I'm blind which is actually kind of a compliment :) running NPM now, I'll ping you on gitter if I get stuck

**Comment by benmcmahon100** [comment]
To be honest i forget what I'm typing as I'm typing it
